### PR TITLE
remove hardcoded rds eol thresholds

### PR DESCRIPTION
**Pull request by ninabauer**
APPSRE-8636

Removes hardcoded rds eol thresholds, and instead gets thresholds from the configmap.

Adds default threshold values when loading config to avoid the alerts flapping and the tickets getting regenerated during the transition.

### CODE REVIEW

File: pkg/config.go [hunk 1]
```diff
@@ -19,8 +19,14 @@ type BaseConfig struct {
...
-	Regions    []string  `yaml:"regions"`
-	EOLInfos   []EOLInfo `yaml:"eol_info"`
+	Regions    []string    `yaml:"regions"`
+	EOLInfos   []EOLInfo   `yaml:"eol_info"`
+	Thresholds []Threshold `yaml:"thresholds"`
+}
+
+type Threshold struct {
+	Name      string `yaml:"name"`
+	Threshold int    `yaml:"threshold"`
```
Comment (janboll) [inline pkg/config.go:29]: Threshold.Threshold? That's confusing :), perhaps Value?

File: pkg/config.go [hunk 2]
```diff
@@ -19,8 +19,14 @@ type BaseConfig struct {
...
 	BaseConfig `yaml:"base,inline"`
-	Regions    []string  `yaml:"regions"`
-	EOLInfos   []EOLInfo `yaml:"eol_info"`
+	Regions    []string    `yaml:"regions"`
+	EOLInfos   []EOLInfo   `yaml:"eol_info"`
+	Thresholds []Threshold `yaml:"thresholds"`
+}
+
+type Threshold struct {
+	Name      string `yaml:"name"`
```
Comment (janboll) [inline pkg/config.go:28]: Any way to validate that name? Thinking a typo could cause missed alerts? WDYT?

File: pkg/config.go (hunk 1 above)
Comment (ninabauer) [reply pkg/config.go:29]: Thresholds makes sense to me. It is the point at which the EOL status will change. In the configmap it will look like:
      thresholds:
        - name: green
          threshold: 365
        - name: yellow
          threshold: 180
        - name: red
          threshold: 90

Comment (ninabauer) [reply pkg/config.go:29]: Nevermind, I see what you mean :)

Comment (ninabauer) [reply pkg/config.go:29]: How about Thresholds.Threshold ?

Comment (hemslo) [reply pkg/config.go:29]: In the config `thresholds` is a list of `Threshold`, each `Threshold` contains 2 fields (name and int), to make it clear that int means day, how about change it to

```yaml
thresholds:
- name: green
  days: 365
```

`Threshold.Days`

File: pkg/rds.go [hunk 3]
```diff
@@ -586,22 +588,32 @@ func (e *RDSExporter) addAllInstanceMetrics(sessionIndex int, instances []*rds.D
...
-		eolStatus = "green"
+
+	for _, threshold := range thresholds {
+		if daysToEOL < threshold.Threshold {
+			return threshold.Name, nil
+		}
+		if threshold.Threshold > highestThreshold {
+			highestThreshold = threshold.Threshold
+			highestThresholdName = threshold.Name
+		}
```
Comment (hemslo) [inline pkg/rds.go:608]: `thresholds` may not order by `Threshold`, to follow the existing logic,
need to first sort by `Threshold`, then return the first name when daysToEOL less than threshold, if not found, return the last name.

Need to add test cases for this.

Comment (hemslo) [reply pkg/rds.go]: some edge cases:

* `daysToEOL` is exactly `thresholds[len(thresholds)-1].Days`
* can `thresholds` be empty?

The expected return value can't reach `"", nil`, and the log is never executed, right?

Comment (ninabauer) [reply pkg/rds.go]: Good catch, thresholds should not ever be empty but I added error handling just in case

**Review by hemslo — APPROVED** [review]
LGTM!
